kstreams/streams.py

```diff
@@ -93,6 +96,18 @@ async def func_wrapper(func):
         await self.consumer.start()
         self.running = True
 
+        if self.initial_offsets is not None:
+            if isinstance(self.topics, List):
+                if len(self.topics) != 1:
+                    raise NotImplementedError(
+                        "Initial offsets with multiple topics is not supported"
+                    )
+
+            for po in self.initial_offsets:
```

Comment:
If you use include the `Topic` name in the initial struct you won't have to iterate twice and you can set initial offsets for multiple topics

Reply:
I could do say initial_offsets is a list of TopicPartitionOffsetRequests, which each contain a topic and a list of partition_offsets (partition, offset).

Or initial offsets is a list of TopicPartitionOffsets, which each contain a topic, partition, and offset.

do you like either of these?

Reply:
going with the second. as discussed offline.